### PR TITLE
For #57: Implementation of Metadata.Maven

### PR DESCRIPTION
**Pull request by paulodamaso**
For #57 :
- Added `Metadata.Maven` test for correct generation
- Changed puzzle to implement correct method

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#59](https://github.com/artipie/maven-adapter/issues/59) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/53972ddb-69ca-47c7-bfdb-10ce4e928303); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/maven/metadata/package-info.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * Artifact abstractions for maven adapter..
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/maven/metadata/package-info.java]: @paulodamaso too many periods (`.`) here at the end.

File: src/test/java/com/artipie/maven/metadata/MetadataMavenTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    @Disabled
+    public void createsMetadata() {
+        MatcherAssert.assertThat(
+            Flowable
+                .fromPublisher(
+                    new Metadata.Maven(
+                        () -> new ListOf(
+                            new FakeFile(
+                                new TextOf("artifact-0.1.jar"),
+                                "jar file 0.1".getBytes()
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/maven/metadata/MetadataMavenTest.java]: @paulodamaso let's use `StandardCharsets.UTF_8` here.

File: src/test/java/com/artipie/maven/metadata/MetadataMavenTest.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                            ),
+                            new FakeFile(
+                                new TextOf("artifact-0.1-source.jar"),
+                                "source file 0.1".getBytes()
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/maven/metadata/MetadataMavenTest.java]: @paulodamaso same here, let's use `StandardCharsets.UTF_8`.

File: src/test/java/com/artipie/maven/metadata/MetadataMavenTest.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                            ),
+                            new FakeFile(
+                                new TextOf("artifact-0.1-javadoc.jar"),
+                                "javadoc file 0.1".getBytes()
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/maven/metadata/MetadataMavenTest.java]: @paulodamaso same here, let's use `StandardCharsets.UTF_8`.

File: src/main/java/com/artipie/maven/metadata/Metadata.java [hunk 5]
```diff
@@ -36,12 +36,12 @@
  *
  * @since 0.2
  *
- * @todo #54:30min Implement Metadata generation.
+ * @todo #57:30min Continue to implement metadata generation.
```
Comment (carlosmiranda) [inline src/main/java/com/artipie/maven/metadata/Metadata.java]: @paulodamaso what metadata is already implemented and what is still missing? Context is missing here, which may be problematic if the next ticket is assigned to someone else.

**Comment by paulodamaso** [comment]
@carlosmiranda Metadata is based on what we have currently at `Artifact`. For now, we just have the file list, as an Artifact very crude abstraction; as we implement new features into `Artifact`, the metadata generation will have to be changed too.

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)[/z](https://www.0crat.com/u/carlosmiranda) didn't make CR comments

**Comment by 0crat** [comment]
The project doesn't have enough funds, can't make a payment

**Pull request merged**